v2/pkg/variablesvalidation/variablesvalidation_test.go

```diff
@@ -16,8 +16,132 @@ import (
 	"github.com/wundergraph/graphql-go-tools/v2/pkg/internal/unsafeparser"
 )
 
-func TestVariablesValidation(t *testing.T) {
+func TestVariablesValidationApolloCompatibility(t *testing.T) {
+	t.Run("extension code is propagated with apollo compatibility flag", func(t *testing.T) {
```

Comment:
Please can you add one test that turns gateway flag on and router flag on and assert the router flag takes precedence?

Reply:
Done